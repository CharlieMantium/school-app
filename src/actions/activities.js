import uuid from 'uuid';

export const addActivity = (
  {
    name = 'Just stuff',
    teacher = 'Anonymous',
    day,
    classNo,
    room = 'Anywhere'
  } = {}
) => ({
  type: 'ADD_ACTIVITY',
  activity: {
    id: uuid(),
    name,
    teacher,
    day,
    classNo,
    room
  }
})

export const removeActivity = ({id} = {}) => ({
  type: 'REMOVE_ACTIVITY',
  id
});

export const editActivity = (id, updates) => ({
  type: 'EDIT_ACTIVITY',
  id,
  updates
});
