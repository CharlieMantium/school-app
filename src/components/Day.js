import React from 'react';
import PropTypes from 'prop-types';
import activityPropTypeShape from '../prop-types/activity';
import ActivitiesListItem from './ActivitiesListItem';

const Day = ({ activities, weekDay }) => (
  <div>
    <p>{weekDay.toUpperCase()}</p>
    {activities.map(activity => (
      <ActivitiesListItem {...activity} key={activity.id} />
    ))}
  </div>
);

Day.defaultProps = {
  activities: [],
};

Day.propTypes = {
  activities: PropTypes.arrayOf(PropTypes.shape(activityPropTypeShape)),
  weekDay: PropTypes.string.isRequired,
};

export default Day;
