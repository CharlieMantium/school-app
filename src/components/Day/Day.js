import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { orderBy } from 'lodash';
import { FormattedMessage } from 'react-intl';

import activityPropTypeShape from 'prop-types/activity';
import { colors, effects, spacing } from 'styles/base';

import ActivitiesListItem from '../ActivitiesListItem';

const DayWrapper = styled.div`
  border: ${spacing.xsSize} solid ${colors.white};
  border-radius: ${spacing.sSize};
  background: rgba(255, 255, 255, 0.8);
  margin: ${spacing.sSize};
  padding: ${spacing.sSize};
  @media (min-width: ${spacing.desktopBreakpoint}) {
    width: 14.2%;
  }
`;

const DayName = styled(FormattedMessage)`
  border-bottom: ${spacing.xxsSize} solid currentColor;
  font-weight: bold;
  margin: ${spacing.sSize} auto;
  text-align: center;
  text-shadow: ${effects.outline(colors.white)};
  width: 70%;
`;

const Day = ({ activities, weekDay }) => (
  <DayWrapper>
    <DayName id="dayName" as="p" data-test="day-name">
      {weekDay.toUpperCase()}
    </DayName>
    {orderBy(activities, activity => activity.activityOrdinal).map(activity => (
      <ActivitiesListItem
        activity={activity}
        key={activity.name}
        data-test="activities-list-item"
      />
    ))}
  </DayWrapper>
);

Day.propTypes = {
  activities: PropTypes.arrayOf(PropTypes.shape(activityPropTypeShape)),
  weekDay: PropTypes.string.isRequired,
};

Day.defaultProps = {
  activities: [],
};

export default Day;
