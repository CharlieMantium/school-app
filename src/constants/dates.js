export const monday = 'monday';
export const tuesday = 'tuesday';
export const wednesday = 'wednesday';
export const thursday = 'thursday';
export const friday = 'friday';
export const saturday = 'saturday';
export const sunday = 'sunday';
export const daysOfTheWeek = [monday, tuesday, wednesday, thursday, friday, saturday, sunday];
