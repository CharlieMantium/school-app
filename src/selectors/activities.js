export default (activities, {text}) => {
  return activities.filter((activity) => {
    const textMatchName = activity.name.toLowerCase().includes(text.toLowerCase());
    const textMatchDay = activity.day.toLowerCase().includes(text.toLowerCase());
    const textMatchTeacher = activity.teacher.toLowerCase().includes(text.toLowerCase());
    const textMatchRoom = activity.room.toLowerCase().includes(text.toLowerCase());
    return textMatchName || textMatchDay || textMatchTeacher || textMatchRoom;
  });
};