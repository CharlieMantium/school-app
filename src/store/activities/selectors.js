export const getActivitiesForDay = (activities, day) => {
	return activities.filter(activity => activity.day === day);
};

export const getActivityItems = state => {
	return state.activities.items;
};

export const getEditedActivity = (state, props) => {
	return state.activities.items.find(
		activity => activity.id === props.match.params.id,
	);
};
