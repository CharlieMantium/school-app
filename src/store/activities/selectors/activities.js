export default (activities, { text }) => {
  return activities.filter(({ name, day, teacher, room }) => {
    const lowerCaseText = text.toLowerCase();
    const lowerCaseName = name.toLowerCase();
    const lowerCaseDay = day.toLowerCase();
    const lowetCaseTeacher = teacher.toLowerCase();
    const lowerCaseRoom = room.toLowerCase();
    const textMatchName = lowerCaseName.includes(lowerCaseText);
    const textMatchDay = lowerCaseDay.includes(lowerCaseText);
    const textMatchTeacher = lowetCaseTeacher.includes(lowerCaseText);
    const textMatchRoom = lowerCaseRoom.includes(lowerCaseText);
    return textMatchName || textMatchDay || textMatchTeacher || textMatchRoom;
  });
};

//TODO: try to create a function, which gets name/day/teacher/ etc
// and text and returns what it should returns.
// Target of this selector is not clear for me. Can you elaborate?
