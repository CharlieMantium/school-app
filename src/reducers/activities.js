import { ADD_ACTIVITY, REMOVE_ACTIVITY, EDIT_ACTIVITY } from '../store/activities/actionTypes';

const activitiesReducerDefaultState = [];

export default (state = activitiesReducerDefaultState, action) => {
  switch (action.type) {
    case ADD_ACTIVITY:
      return [...state, action.activity];
    case REMOVE_ACTIVITY:
      return state.filter(activity => activity.id !== action.id);
    case EDIT_ACTIVITY:
      return state.map(activity => {
        if (activity.id === action.id) {
          return {
            ...activity,
            ...action.updates,
          };
        }
        return activity;
      });
    default:
      return state;
  }
};

// TODO: Imo state shuould be an object
// Probably you will have more fields here in this object.
