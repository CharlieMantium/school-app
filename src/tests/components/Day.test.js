import React from 'react';
import { shallow } from 'enzyme';

import Day from '../../components/Day';
import testState from '../fixtures/state';
import { getActivitiesForDay } from '../../store/activities/selectors';
import { MONDAY } from '../../constants/dates';

test('should render Day with testState', () => {
  const testDay = MONDAY;
  const wrapper = shallow(
    <Day weekDay={testDay} activities={getActivitiesForDay(testState.activities.items, testDay)} />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render Day with no activities', () => {
  const testDay = MONDAY;
  const wrapper = shallow(<Day weekDay={testDay} activities={getActivitiesForDay([], testDay)} />);
  expect(wrapper).toMatchSnapshot();
});
