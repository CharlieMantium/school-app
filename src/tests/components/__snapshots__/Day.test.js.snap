// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Day with no activities 1`] = `
<div>
  <p>
    MONDAY
  </p>
</div>
`;

exports[`should render Day with testState 1`] = `
<div>
  <p>
    MONDAY
  </p>
  <ActivitiesListItem
    activity={
      Object {
        "classNo": "1",
        "day": "monday",
        "id": "2",
        "name": "polski",
        "room": "hell",
        "teacher": "Malski",
      }
    }
    key="2"
  />
  <ActivitiesListItem
    activity={
      Object {
        "classNo": "2",
        "day": "monday",
        "id": "3",
        "name": "polski",
        "room": "103",
        "teacher": "Malski",
      }
    }
    key="3"
  />
</div>
`;
